# DeepEnyme_score : Identification of Distal Allosteric Mutations for modulating Enzyme functions 

Here we present an approach for identifying and screening distal mutations sites of enzymes for enhanced activity. The approach integrates **dynamic cross-correlation matrix (DCCM) analysis**, and** machine-learning–assisted ranking** to rationally screen distal variants for tailored enzyme function.

---

## Overview

Traditional enzyme engineering strategies primarily focus on residues around the active site. However, increasing evidence suggests that **long-range dynamic coupling and allosteric communication** play a critical role in catalysis.


---

## Workflow Summary
<img width="1810" height="427" alt="image" src="https://github.com/user-attachments/assets/a1bcf1c8-2032-47b3-9cb1-94e1856e58cb" />


---

## Dependencies

**External tools:**

Boltz2 (https://github.com/jwohlwend/boltz)

DLKcat (https://github.com/SysBioChalmers/DLKcat)

AmberTools (cpptraj)

---

## Input Data

The Machine learning pipeline is based on **MD trajectories**


---

## Step 1: MD Trajectory Processing

> **Note:** MD simulations are performed using standard MD software packages (e.g. AMBER, GROMACS).

---

## Step 2: DCCM and ΔDCCM Analysis

Dynamic cross-correlation matrices (DCCMs) were generated using the `cpptraj` module of AmberTools.  
For each MD trajectory, the following command was used:

```bash
matrix correl name DCCM @CA out dccm.dat
```
This command computes Cα–Cα dynamic cross-correlation coefficients and outputs the DCCM as a plain-text matrix (dccm.dat).

The DCCM file of the enzyme species were post-processed using Python scripts provided in this repository to identify residues dynamically coupled to a specified reference position, and the results are saved as a CSV file:

```
python scripts/DCCM-csv.py \
  --wt data/dccm/WT.dat \
  --mut data/dccm/mutant.dat \
  --ref-res 42 \
  --threshold 0.3 \
  --out results/DCCM_mutant.csv
```


---

## Step 3: Focused Mutation Library Design

Run the following scripts to obtain the non-conserved position information.
```
python scripts/msa_consensus.py \
  --aln data/MSA/and_varinats.aln \
  --out results/msa_consensus.txt
```

> The `.aln` file required for MSA analysis is generated by `BLAST`.

Then for each retained position:
- The **most frequently observed residue at that position** in the MSA
```
python scripts/design_mutations.py \
  --positions results/DCCM_mutant.csv \
  --consensus results/msa_consensus.txt \
  --wt-seq data/sequences/WT.fasta \
  --out results/mutation_list.txt
```

> The `.fasta` file is the sequence for wild type enzyme is chosen as the amino acid to be mutated into.

New sequences are generated for downstream deep-learning based modeling and scoring task 

```
python scripts/generate_mutant_sequences.py \
  --wt-seq data/sequences/WT.fasta \
  --mutations results/mutation_list.txt \
  --out results/single_mutant_library.csv
```

---

## Step 4: Structure Modelling (Boltz2)

The structure of each sequence in complex with substrate is modelled using the **Boltz2 modelling protocol**.

Catalytic turnover numbers (`kcat`) will be predicted using **DLKcat**.

> Boltz2 and DLKcat are an external tools and are **not implemented in this repository**.


Variants are ranked by a composite score that we designed, integrating both binding affinity and catalytic efficiency of enzymes,

DeepEnzyme_Score:
$\`
DeepEnzyme score = kcat / (1 - Affinity score)
\`$

**Output:**

Ranked list of candidate distal mutations.


## Directory Structure

```
├── README.md
├── scripts/
│   ├── compute_delta_dccm.py
│   ├── identify_perturbed_residues.py
│   ├── filter_by_msa.py
│   ├── build_mutation_library.py
│   └── rank_variants.py
├── data/
│   ├── md/
│   ├── dccm/
│   └── msa/
├── results/
│   ├── candidate_residues.txt
│   ├── mutation_library.csv
│   └── ranked_variants.csv
└── environment.yml
```
## Note

This Github project code are demonstrated on: Terpene synthases 

The associated manuscript is currently under review.

## Contact
Jiahui Zhou (jiahui.zhou@qub.ac.uk) Queen's University Belfast, UK

Meilan Huang (m.huang@qub.ac.uk) Queen's University Belfast, UK

https://www.huanggroup.co.uk/
